modules/programs/gradle.nix

```
@@ -0,0 +1,134 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.gradle;
+  defaultHomeDirectory = ".gradle";
+  settingsFormat = pkgs.formats.javaProperties { };
+
+  initScript = types.submodule ({ config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = ''
+          The name of the init script. Has to end with either '.gradle' for
+          Groovy init scripts or '.init.gradle.kts' for Kotlin init scripts.
+          See https://docs.gradle.org/current/userguide/init_scripts.html for
+          more details.
+        '';
+      };
+      text = mkOption {
+        type = types.lines;
+        default =
+          if config.source != null then builtins.readFile config.source else "";
+        defaultText = literalExpression
+          "if source is defined, the content of source, otherwise empty";
+        description = ''
+          Text of the init script file.
+          If unset then the source option will be preferred.
+        '';
```

This is too complicated w.r.t merging and will cause confusion, and likely infinite recursion in some cases. I would suggest using a simpler setup similar to what is in `file-type.nix`:

```suggestion
        type = types.nullOr types.lines;
        default = null;
        description = ''
          Text of the init script file.
          If unset then the source option must be specified.
        '';
```

---

I adopted this from the [nushell module](https://github.com/nix-community/home-manager/blob/948703f3e71f1332a0cb535ebaf5cb14946e3724/modules/programs/nushell.nix#L64-L82) as suggested [here](https://github.com/nix-community/home-manager/pull/4667#discussion_r1413594398). Maybe that also needs to be fixed?